CustomUser DAO and Top DAO to manage the database.

Mefjuu94 opened the pull request:
CustomUser DAO and Top DAO classes were created to manage the database.
have methods that allow you to:
- save a new object
- load an object by a given name
- find an object with specific attributes
- delete an object by a given name
Tests were also created for each method.

Mefjuu94 committed:
Created classes:
- "Top" who represents Summit with fields:
* name
* height
* latitude;
* longitude;
- "TopDAO" is class to maintenance database with following methods:
 * addTop
 * getAllSummits
 * findSummitByName -> returns list of Summits because for Ex. "Rysy" are in Slovakia (2503 height) and in Polish side (2499 height)
 * findSummitByHeight
- "PeaksAndSummitsHandler" who reading file from resource folder with summits data, sort it and adds as new Java Object and add to static list.
- "UserSessionFactory" to create Session for database.

Mefjuu94 committed:
- add methods to "TopDAO":
 * findSummitsByHeightGreaterThan
 * findSummitsByHeightLessThan
 * findSummitsByHeightBetween
 * deleteTop

- Added Tests for "TopDAO":
(happy path:)
* addTopTest
* findSummitTest
* getAllSummitRecordsNumberTest
* deleteSummitTest
* findSummitHeightBetween999_1000Test
(unhappy path:)
* findSummitHeightBetween999_1000TestFail
* findSummitTestFail

Mefjuu94 committed:
*Added:
- Login Page
- Register Page
- User Page
- Start Page (index)
Added Classes:
- AppConfig
- CustomUser
- CustomUserDAO
- CustomUserSecurityService
- RestController
- SecurityController
- SecurityConfig
- User

Mefjuu94 committed:
*Added:
- Login Page
- Register Page
- User Page
- Start Page (index)
Added Classes:
- AppConfig
- CustomUser
- CustomUserDAO
- CustomUserSecurityService
- RestController
- SecurityController
- SecurityConfig
- User

Mefjuu94 committed:
Merge remote-tracking branch 'origin/TopDAO' into TopDAO

Mefjuu94 committed:
*Modified:
- Login Page
- Register Page
- Start Page (index)
Added:
- few background images
- css files for login, register and start Pages
-

Mefjuu94 committed:
*Modified:
- Login Page
- Register Page
- Start Page (index)
Added:
- few background images
- css files for login, register and start Pages

Mefjuu94 committed:
*Added Tests for TopDAO.
- addBigTopTestFail ->too big number should return false
- addSmallTopTestFail ->too small number should return false
- getAllSummitRecordsNumberTest -> should have 7912 records
- deleteSummitTestFail -> no object with name "testing" should return false
- findSummitHeightBetween0_2TestFail -> no object with that hight should return empty list
- findSummitsByHeightGreaterThan -> should return true
- findSummitsByHeightGreaterThanFail -> no object with that hight should return empty list
- findSummitsByHeightGreaterThan8000isEmpty -> no object with that hight should return empty list
- findSummitsByHeightLessThan -> should return true
- findSummitsByHeightLessThan2isEmpty-> should return nop result (empty list)
- findSummitsByHeightLessThanFail -> should return nop result (empty list)
- findSummitsByHeight
- findSummitsByHeightNoResult -> should return nop result (empty list)
*Added Tests for CustomUserDAO.
- saveCustomUserTest -> should return true
- saveCustomUserTestFail -> already exist user with this email should return false
- findCustomUserTest -> should return true
- findCustomUserTestFail -> no user with this email should return false
- deleteCustomUserTest -> should return true (comparison via email due to automatic ID change)
- deleteCustomUserTestFail -> no user with this email should return false
*deleted folder "beans" with ununnecessary classes.

blazejprzyluski commented on build.gradle:
```diff
@@ -25,6 +25,9 @@ dependencies {
     implementation 'org.springframework.boot:spring-boot-starter-web'
     implementation 'org.thymeleaf.extras:thymeleaf-extras-springsecurity6'
+
+    implementation 'org.apache.poi:poi:5.2.5'
```
What is the purpose of this dependancy?

Mefjuu94 replied:
help fulfill the database from the XML file located in resources/peaksAndSummits/Wierzcholki-wzgorza-i-szczyty-gor-w-Polsce.xls
With first run, or if something going wrong always can use it to make "clear" database with Summits.

blazejprzyluski commented on src/main/java/TTT/controller/RestController.java:
```diff
@@ -0,0 +1,19 @@
+import org.springframework.web.bind.annotation.PostMapping;
+
+@org.springframework.web.bind.annotation.RestController
+public class RestController {
```
Are you intending to add REST API for custom users or is it a leftover from studying?

Mefjuu94 replied:
leftover from science. my oversight.

blazejprzyluski commented on src/main/java/TTT/controller/SecurityController.java:
```diff
@@ -0,0 +1,36 @@
+        if (bindingResult.hasErrors()){
+            return "security/register";
+        }
+        dao.saveUser(customUser);
```
Perhaps more validation of customUser could be added?

Mefjuu94 replied:
sure!

blazejprzyluski commented on src/main/java/TTT/controller/UserController.java:
```diff
@@ -0,0 +1,26 @@
+import org.springframework.web.bind.annotation.GetMapping;
+
+@Controller
+public class UserController {
```
Why is user controller handling both main page and user details?

Mefjuu94 replied:
My bad. Corrected!

blazejprzyluski commented on src/main/java/TTT/databaseUtils/CustomUserDAO.java:
```diff
@@ -0,0 +1,70 @@
+            session.close();
+            return true;
+        } else {
+            System.out.println("EntityExistsException: email alerady exist.");
```
Better to throw an exception, you can't do anything with console print

Mefjuu94 replied:
even if I want to add a user like exist by manually setting his ID, hibernate will change it, not allowing duplication. Therefore, No entityException occurs. Print, however, I simply copied it to know exactly what I mean.

blazejprzyluski commented on src/main/java/TTT/databaseUtils/CustomUserDAO.java:
```diff
@@ -0,0 +1,70 @@
+            CustomUser results = session.createQuery(userQuery).getSingleResult();
+            return results;
+        } catch (NoResultException e) {
+            System.out.println("there is no Customer User with that email");
```
Better to throw an exception

Mefjuu94 replied:
that is better?
throw new NoResultException("No entity found with email: " + email);

blazejprzyluski commented on src/main/java/TTT/databaseUtils/CustomUserDAO.java:
```diff
@@ -0,0 +1,70 @@
+            session.getTransaction().commit();
+            return true;
+        } else {
+            System.out.println("there is no Customer User with that email");
```
Better to throw an exception

Mefjuu94 replied:
catch (PersistenceException | IllegalArgumentException e) {
throw new RuntimeException("there is no Customer User with that email");
seems ok?

blazejprzyluski commented on src/main/java/TTT/databaseUtils/TopDAO.java:
```diff
@@ -0,0 +1,134 @@
+    private final SessionFactory sessionFactory = UserSessionFactory.getUserSessionFactory();
+
+
+    public boolean addTop(Top top) {
```
addSummit

Mefjuu94 replied:
Changed.

blazejprzyluski commented on src/main/java/TTT/databaseUtils/TopDAO.java:
```diff
@@ -0,0 +1,134 @@
+            session.getTransaction().commit();
+            return true;
+        } else {
+            System.out.println("No summit with that name");
```
I'd suggest throwing an exception

blazejprzyluski commented on src/main/java/TTT/peaksAndSummitsHandler/PeaksAndSummitsHandler.java:
```diff
@@ -0,0 +1,69 @@
+public class PeaksAndSummitsHandler {
+    public static ArrayList<Top> polishSummits = new ArrayList<>();
+
+    public PeaksAndSummitsHandler() {
```
Why the empty constructor?

Mefjuu94 replied:
remnant of initial sketch. Deleted.

blazejprzyluski commented on src/main/java/TTT/peaksAndSummitsHandler/PeaksAndSummitsHandler.java:
```diff
@@ -0,0 +1,69 @@
+import java.util.ArrayList;
+
+public class PeaksAndSummitsHandler {
+    public static ArrayList<Top> polishSummits = new ArrayList<>();
```
It seems like you're trying to create a singleton here, but this is not how you do it

blazejprzyluski commented on src/main/java/TTT/users/AppConfig.java:
```diff
@@ -0,0 +1,13 @@
+public class AppConfig {
+
+    @Bean
+    public CustomUser defaultUser(){
```
Are you using this class?

Mefjuu94 replied:
No...

blazejprzyluski commented on src/test/java/CustomUserDAOTests.java:
```diff
@@ -0,0 +1,48 @@
+    @Test
+    public void saveCustomUserTest(){
+        Assertions.assertTrue(testObject.saveUser(testCustomUser));
+        testObject.deleteCustomUser("test@mail.com");
```
Can you also test if the user was deleted?

Mefjuu94 replied:
This is to make sure that no unnecessary objects are left in the database after this action. Because after testing objects do NOT disappear.

blazejprzyluski commented on src/test/java/CustomUserDAOTests.java:
```diff
@@ -0,0 +1,48 @@
+
+public class CustomUserDAOTests {
+
+    CustomUserDAO testObject = new CustomUserDAO();
```
private

Mefjuu94 replied:
Ok.

blazejprzyluski commented on src/test/java/CustomUserDAOTests.java:
```diff
@@ -0,0 +1,48 @@
+public class CustomUserDAOTests {
+
+    CustomUserDAO testObject = new CustomUserDAO();
+    CustomUser testCustomUser = new CustomUser("test@mail.com","123");
```
private

Mefjuu94 replied:
Done.

blazejprzyluski commented on src/test/java/CustomUserDAOTests.java:
```diff
@@ -0,0 +1,48 @@
+    @Test
+    public void saveCustomUserTestFail(){
+        testObject.saveUser(testCustomUser);
+        Assertions.assertFalse(testObject.saveUser(testCustomUser));
```
why does it fail?

Mefjuu94 replied:
because user with the same email exist, and returns false like supposed.

blazejprzyluski commented on src/test/java/TopDaoTests.java:
```diff
@@ -0,0 +1,145 @@
+
+public class TopDaoTests {
+
+    TopDAO testObject = new TopDAO();
+    Top testTop = new Top("test", 1, 111, 222);
+    List<Top> testlist = new ArrayList<>();
```
make private

Mefjuu94 replied:
Done.

blazejprzyluski commented on src/test/java/TopDaoTests.java:
```diff
@@ -0,0 +1,145 @@
+import java.util.List;
+
+
+public class TopDaoTests {
```
Those tests do not seem to be able to run independantly, rather all at once/in order. Please introduce tests in such a way that they're not dependant on each other.

Mefjuu94 replied:
Ok, do some changes, please check it.

Mefjuu94 committed:
*Fixed All tests from CustomUser
*Deleted unnecessary classes
*Modified CustomUserDAO
*Added more validation to CustomUserDAO
